middleware.go

```diff
@@ -62,34 +63,42 @@ func (traceConfig *TraceConfig) TraceMiddleware(delegate http.Handler) http.Hand
 	})
 }
 
-// EchoFirstNodeTraceInfo captures the trace information from a request and writes it
-// back in the response headers if the request is the first one in the trace path.
-func EchoFirstTraceNodeInfo(propagator propagation.TextMapPropagator) func(http.Handler) http.Handler {
+// EchoFirstNodeTraceInfo captures the trace information from a request, writes it
+// back in the response headers, and adds it to the request's context
+// It can also decode the request and save the resulting WRP object in the context if isDecodable is true
+func EchoFirstTraceNodeInfo(propagator propagation.TextMapPropagator, isDecodable bool) func(http.Handler) http.Handler {
```

Comment:
For future:

This is the kind of function that will probably grow over time, with new parameters.  This makes the API brittle, since you wind up breaking existing clients when you change the function's signature.

If we add more parameters here, it would probably be better to move to either the builder pattern or the functional options pattern described by https://golang.cafe/blog/golang-functional-options-pattern.html.

An really trivial example:

```
type TraceOption func(*traceHandlerBuilder)

type traceHandlerBuilder {
    // whatever fields are necessary to build the handler
}

func (builder traceHandlerBuilder) build() func(http.Handler) http.Handler { ... }

func EchoFirstTraceNodeInfo(opts ...TraceOption) func(http.Handler) http.Handler {
   var builder traceHandlerBuilder
   for _, o := range opts {
        o(&builder)
   }

   return o.build()
}
```

Alternatively, you can expose the builder as part of your API.  `gorilla/mux` takes this approach.  The choice of which is really more about what will be more stable for a client.

Reply:
Created a ticket for this work: #142